homeassistant/components/iqvia/__init__.py

```diff
@@ -251,7 +307,25 @@ def update():
             self.hass, TOPIC_DATA_UPDATE, update
         )
 
+        await self._iqvia.async_register_api_interest(self._type)
+        if self._type == TYPE_ALLERGY_FORECAST:
+            # Entities that express interest in allergy forecast data should also
+            # express interest in allergy outlook data:
+            await self._iqvia.async_register_api_interest(TYPE_ALLERGY_OUTLOOK)
+
+        await self.async_update()
```

Comment:
`async_added_to_hass` should not trigger an update.

Reply:
Without this, the entities show `unavailable` until their next update cycle. With this "API registration" mechanism in play, I'm unclear on how to instruct an entity to update immediately after its data appears without this.

Reply:
@balloob How do you feel about https://github.com/home-assistant/core/pull/32291/commits/ac1b9484c8ff1a8fd85fb69e471b18f7fb474b28?

Reply:
@balloob How do you feel about c41edf1ae363?